rust_src/src/lists.rs

```diff
@@ -240,15 +240,7 @@ impl From<LispObject> for LispCons {
     }
 }
 
-impl From<LispObject> for Option<LispCons> {
-    fn from(o: LispObject) -> Self {
-        if o.is_list() {
```

Comment:
Also a change in behavior. Here if `as_cons()` was called on a fixnum the result would be None. Now it is an error.